OpenRA.Mods.AS/Traits/World/TintedCellsLayer.cs

```diff
@@ -0,0 +1,212 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2015- OpenRA.Mods.AS Developers (see AUTHORS)
+ * This file is a part of a third-party plugin for OpenRA, which is
+ * free software. It is made available to you under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation. For more information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Drawing;
+using OpenRA.Graphics;
+using OpenRA.Mods.AS.Graphics;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.AS.Traits
+{
+	[Desc("Has to be attached to world actor. ")]
+	public class TintedCellsLayerInfo : ITraitInfo, IRulesetLoaded
+	{
+		[Desc("Color of cells")]
+		public readonly Color Color = Color.FromArgb(0, 255, 0);
+
+		[Desc("Maximum radiation allowable in a cell.The cell can actually have more radiation but it will only damage as if it had the maximum level.")]
+		public readonly int MaxLevel = 500;
+
+		[Desc("Delay in ticks between level decrements. The level updates this often, although the whole lifetime will be as defined by half-life.")]
+		public readonly int UpdateDelay = 15;
+
+		[Desc("The alpha value for displaying level for cells with level == 1")]
+		public readonly int Darkest = 4;
+
+		[Desc("The alpha value for displaying level for cells with level == MaxLevel")]
+		public readonly int Brightest = 64;
+
+		[Desc("Delay of half life, in ticks")]
+		public readonly int Halflife = 150;
+
+		[Desc("Z offset of the visualization.")]
+		public readonly int ZOffset = -10;
+
+		[Desc("Name of the layer.")]
+		public readonly string Name = "radioactivity";
+
+		[Desc("Damage received per level, in per mille, per DamageInterval. (Damage = DamageCoeff * RadioactivityLevel / 1000")]
```

Comment:
Shifting these here was a bad idea. Having them per-unit was better. Also DamageCoeff?

Reply:
Per-trait variants even allow to create decreasing rad affecttion with conditioons, sooo....